src/CheckIfDead.php

```diff
@@ -378,15 +377,52 @@ protected function sanitizeURL( $url ) {
 			);
 		}
 		if ( isset( $parts['query'] ) ) {
-			$url .= "?".urlencode( urldecode( $parts['query'] ) );
+			$url .= "?";
+			$parts['query'] = explode( '&', $parts['query'] );
```

Comment:
Can you leave a few comments here about what's going on? What does <code>$url .= "?"'</code> do? What does L381 do?